lib/active_record_shards/tasks.rb

```diff
@@ -4,16 +4,22 @@
   Rake::Task[name].clear
 end
 
+def env_name
+  defined?(Rails.env) ? Rails.env : RAILS_ENV || 'development'
+end
+
+def root_connection(conf)
+  ActiveRecord::Base.send("#{conf['adapter']}_connection", conf.merge('database' => nil))
+end
+
 namespace :db do
   desc 'Drops the database for the current RAILS_ENV including shards'
   task :drop => :load_config do
-    env_name = defined?(Rails.env) ? Rails.env : RAILS_ENV || 'development'
     ActiveRecord::Base.configurations.each do |key, conf|
       if key.starts_with?(env_name) && !key.ends_with?("_slave")
         begin
           if ActiveRecord::VERSION::MAJOR >= 4
-            connection = ActiveRecord::Base.send("#{conf['adapter']}_connection", conf.merge('database' => nil))
-            connection.drop_database(conf['database'])
+            root_connection(conf).drop_database(conf['database'])
```

Comment:
ah, nice approach.  I've banged my head on the right way to do this without completely side-stepping active record, this is pretty decent.  :+1:  after grosser's review fixes.